modules/prediction/network/net_model.h

```diff
@@ -75,8 +75,7 @@ class NetModel {
   virtual void State(std::vector<Eigen::MatrixXf>* states) const {}
 
   /**
-   * @brief Set the internal state of a model
-   * @param A specified internal state in a vector of Eigen::MatrixXf
+   * @brief Reset the internal state of a network model
    */
   virtual void ResetState() const {}
```

Comment:
it's better to change those virtual functions to pure virtual functions

Reply:
Agreed.Will include this change in my next commit.Thanks!

Reply:
```AutotuningMLPModel``` implements NetModel but does not override ResetState,State and SetState.
We need to implement these methods for that class and write unit tests before declaring these methods as pure virtual functions in the base class.
Looking into it.